src/contracts/LidoLiquidityManager.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {OwnableOperable} from "./OwnableOperable.sol";
+import {IERC20, IWETH, IStETHWithdrawal} from "./Interfaces.sol";
+
+/**
+ * @title Manages stETH liquidity against the Lido Withdrawal Queue.
+ * @author Origin Protocol Inc
+ */
+abstract contract LidoLiquidityManager is OwnableOperable {
+    IERC20 public immutable steth;
+    IWETH public immutable weth;
+    IStETHWithdrawal public immutable withdrawalQueue;
+
+    uint256 public outstandingEther;
+
+    uint256[50] private _gap;
+
+    constructor(address _steth, address _weth, address _lidoWithdrawalQueue) {
+        steth = IERC20(_steth);
+        weth = IWETH(_weth);
+        withdrawalQueue = IStETHWithdrawal(_lidoWithdrawalQueue);
+    }
+
+    /**
+     * @notice Approve the stETH withdrawal contract. Used for redemption requests.
+     */
+    function approveStETH() external onlyOperatorOrOwner {
+        steth.approve(address(withdrawalQueue), type(uint256).max);
+    }
+
+    /**
+     * @notice Request a stETH for ETH withdrawal.
+     * Reference: https://docs.lido.fi/contracts/withdrawal-queue-erc721/
+     * Note: There is a 1k amount limit. Caller should split large withdrawals in chunks of less or equal to 1k each.)
+     */
+    function requestStETHWithdrawalForETH(uint256[] memory amounts)
+        external
+        onlyOperatorOrOwner
+        returns (uint256[] memory requestIds)
+    {
+        requestIds = withdrawalQueue.requestWithdrawals(amounts, address(this));
+
+        // Increase the Ether outstanding from the Lido Withdrawal Queue
+        for (uint256 i = 0; i < amounts.length; i++) {
+            outstandingEther += amounts[i];
```

Comment:
This gets to be a multiple writes as we get into bigger amounts.

Perhaps we should:
- look our stETH balance before and after
- compare to the sum of amount (revert if not almost exact)
- store the amount one time.

Reply:
Storing the amount one time makes sense. I've done that change.

If we weren't dealing with stETH then I'd agree on implementing the before and after balance check of stETH. The problem is stETH can leave dust amounts behind on transfers so we'd have to include some tolerance for that. It's possible to do so can add if we think it's needed.